cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -727,17 +727,14 @@ std::vector<size_t> reader::impl::calculate_output_num_rows_per_source(size_t co
   // Binary search start_row and end_row in exclusive_sum_num_rows_per_source vector
   auto const start_iter =
     std::upper_bound(partial_sum_nrows_source.cbegin(), partial_sum_nrows_source.cend(), start_row);
-  auto const end_iter =
-    (end_row == _file_itm_data.global_skip_rows + _file_itm_data.global_num_rows)
-      ? partial_sum_nrows_source.cend() - 1
```

Comment:
This can never be true as end_row doesn't include `_file_itm_data.global_skip_rows` anyway. Also, even if we are reading `end_row = skip_rows + num_rows`, this doesn't mean that `end_row` is located in the last source